src/imports/gallery/qdeclarativedocumentgallery.cpp

```diff
@@ -80,7 +82,18 @@ QAbstractGallery *QDeclarativeDocumentGallery::gallery(QObject *object)
         }
     }
 #endif
-    return qt_declarativeDocumentGalleryInstance();
+    if (instance) {
+        return instance;
+    } else {
+        QCoreApplication *app = QCoreApplication::instance();
+        if (!app) {
+            qWarning() << "No QCoreApplication, refusing to create QDocumentGallery instance for QML";
+            return nullptr;
+        } else {
+            instance = new QDocumentGallery(app);
+            return instance;
```

Comment:
nit: stacked else-after-jump thing that does not exactly help readability, at least for me